Public/Src/FrontEnd/Script/Ambients/Transformers/AmbientTransformer.ReadPipGraphFragment.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using BuildXL.FrontEnd.Script.Evaluator;
+using BuildXL.FrontEnd.Script.Types;
+using BuildXL.FrontEnd.Script.Values;
+using static BuildXL.Utilities.FormattableStringEx;
+
+namespace BuildXL.FrontEnd.Script.Ambients.Transformers
+{
+    /// <summary>
+    /// Ambient definition for namespace Transformer.
+    /// </summary>
+    public partial class AmbientTransformerBase : AmbientDefinitionBase
+    {
+        internal const string ReadGraphFragmentFunctionName = "readPipGraphFragment";
+
+        private CallSignature ReadGraphFragmentSignature => CreateSignature(
+            required: RequiredParameters(AmbientTypes.StringType, AmbientTypes.PathType, AmbientTypes.ArrayType),
+            returnType: AmbientTypes.StringType);
+
+        private static int uniqueFragmentId = 0;
+
+        private static EvaluationResult ReadGraphFragment(Context context, ModuleLiteral env, EvaluationStackFrame args)
+        {
+            var file = Args.AsFile(args, 0);
+            var deps = Args.AsArrayLiteral(args, 1).Values.Select(v => (int)v.Value).ToArray();

```

Comment:
>deps [](start = 16, length = 4)

Please validate that the deps are valid FullSymbols. (i.e. valid symbols in the frontend). 
This will allow fragements to participate in the hybrid engine where outputs can be shared.

Reply:
They are now integers that are checked using type checking.

---
In reply to: [289934173](https://github.com/microsoft/BuildXL/pull/394#discussion_r289934173) [](ancestors = 289934173)